test/integration/targets/inventory_aws_conformance/lib/boto/mocks/instances.py

```diff
@@ -179,7 +179,7 @@ class InstanceBase(object):
     def __init__(self, stopped=False):
         # set common ignored attribute to make sure instances have identical tags and security groups
         self._ignore_security_groups = {
-            'sg-0e1d2bd02b45b712e': 'sgname-with-hyphens',
+            'sg-0e1d2bd02b45b712e': 'a-sgname-with-hyphens',
```

Comment:
what's this change for?

Reply:
The contrib script can't handle sorting so it needs to be in the mock instead... but I missed committing the line in BotoInstance that makes this make sense, i.e.
```
diff --git a/test/integration/targets/inventory_aws_conformance/lib/boto/mocks/instances.py b/test/integration/targets/inventory_aws_conformance/lib/boto/mocks/instances.py
index aa2f45dbf0..1ec6a4332e 100644
--- a/test/integration/targets/inventory_aws_conformance/lib/boto/mocks/instances.py
+++ b/test/integration/targets/inventory_aws_conformance/lib/boto/mocks/instances.py
@@ -245,7 +245,7 @@ class BotoInstance(InstanceBase):
         self.dns_name = 'ec2-12-3-456-78.compute-1.amazonaws.com'
         self.eventsSet = None
         self.group_name = None
-        self.groups = [SecurityGroup(k, v) for k, v in self._ignore_security_groups.items()]
+        self.groups = [SecurityGroup(k, self._ignore_security_groups[k]) for k in sorted(self._ignore_security_groups.keys())]
```
After that both the constructed hostvars for and those returned by the script are in a consistent order.

Reply:
https://github.com/ansible/ansible/pull/54423